ros_buildfarm/aggregate_status.py

```diff
@@ -0,0 +1,291 @@
+import os
+import re
+import yaml
+import requests
+import collections
+from .config import get_index as get_config_index, get_release_build_files
+
+"""
+
+  Aggregate Status takes the yaml generated by the different CPU architectures (e.g. ros_lunar_default, ros_lunar_ds)
+  and combines them into one yaml file per ROS distro. It also creates an English description of each package across
+  all of the architectures. See examples below.
+
+  Naming Conventions
+   * distro refers to ROS Distro (e.g. indigo)
+   * machine refers to different release builds (e.g. default, uxhf, dsv8)
+   * os_name refers to the name of the Operating System (e.g. ubuntu, debian)
+   * os_flavor refers to specific distros of the os (e.g. xenial, jessie)
+   * cpu refers to the cpu architecture (or source), (e.g. i386, amd64, source)
+   * candidate refers to different candidate builds, (e.g. build, test, main)
+   * combo refers to the os_flavor + the cpu
```

Comment:
These "naming conventions" sound confusing to me. To be consistent with the existing terms in use I suggest the following mapping:

* `distro` -> `rosdistro`
* `machine` -> `build_file_name`
* `os_flavor` -> `os_code_name`
* `cpu` -> `arch`
* `candidate` -> `repository_name`
* `combo` -> `target`

Reply:
I found the naming of all these things confusing, so I appreciate you pointing me right. The only one I have trouble with is `candidate` -> `repository_name`. It doesn't seem like build, test and main are the names of (git?) repositories, so I was thinking of it more as build, test and main being different candidates for release. Is there another name besides repository name that suits you?

Reply:
They are `apt` repositories.